eclass/cmake.eclass

```diff
@@ -791,11 +651,6 @@ cmake_src_configure() {
 		echo 'set(CMAKE_COLOR_MAKEFILE OFF CACHE BOOL "pretty colors during make" FORCE)' >> "${common_config}" || die
 	fi
 
-	# See bug 735820
-	if [[ ${EAPI} != 7 ]]; then
-		echo 'set(CMAKE_INSTALL_ALWAYS 1)' >> "${common_config}" || die
-	fi
-
 	# Wipe the default optimization flags out of CMake
 	if [[ ${CMAKE_BUILD_TYPE} != Gentoo ]]; then
```

Comment:
@mgorny Did we forget to change this to `RelWithDebInfo` in commit 7e262b0401cd2bb3bf93064c74cb0207b8f31293 and do we want to rectify it with EAPI-9 or earlier?

Reply:
I don't understand what would be wrong with this code. My understanding is that this has `!= Gentoo` because there are no default optimization flags for `Gentoo` build type.

Reply:
I thought we wanted to entirely get rid of the `Gentoo` build type with <EAPI-8, after all it isn't mentioned in docs either. Right now eclassdoc suggests that our new default `RelWithDebInfo` is least likely to append undesireable flags, and to manually patch CMake files if still necessary, but then goes on to wipe out optimisation flags in this code section anyway.

Reply:
@atupone do you remember the reason for the sci-mathematics/cvc4 commit linked above?

Reply:
That should be changed upstream, ideally.

Also, two dev-build/cmake patches making use of it:
[`cmake-3.27.0_rc1-0004-Ensure-that-the-correct-version-of-Qt-is-always-used.patch`](https://github.com/gentoo/gentoo/blob/master/dev-build/cmake/files/cmake-3.27.0_rc1-0004-Ensure-that-the-correct-version-of-Qt-is-always-used.patch)
[`cmake-3.27.0_rc1-0005-Respect-Gentoo-s-Python-eclasses.patch`](https://github.com/gentoo/gentoo/blob/master/dev-build/cmake/files/cmake-3.27.0_rc1-0005-Respect-Gentoo-s-Python-eclasses.patch)